dell/csi-baremetal-operator
# [ISSUE-644] Support non default CSI deployment at Openshift platform

## Purpose
### Issue dell/csi-baremetal#644

Supported validation csi-node-sa and csi-baremetal-extender-sa for priveleged SCC:
 * In case of absence operator will not deploy node and scheduler extender daemonsets and write events about it;
 * When privileged SCC will appear for the service accounts bellow, operator will create node and scheduler daemonsets

## PR checklist

## Testing
1. Deploy CSI operator at non default namespace:
```bash
helm install csi-baremetal-operator csi-baremetal-operator-1.1.0.tgz --set image.tag=1.1.0-105.791e738 --set global.regist
ry=asdrepo.isus.emc.com:9042 --set log.level=debug -n test-csi
```
2. Deploy CSI deployment at non default namespace:
```bash
helm install csi-baremetal csi-baremetal-deployment-1.1.0.tgz --set image.tag=1.1.0-506.dfec944 --set global.registry=asdrepo.isus.emc.com:9042 --set log.level=debug --set scheduler.patcher.enable=true --set scheduler.log.level=debug --set nodeController.log.level=debug --set driver.log.level=debug --set feature.extender=true  --set feature.usenodeannotation=True --set platform=openshift --set logReceiver.create=true --set log.format=text --set driver.drivemgr.type=halmgr -n test-csi
```
3. Check the pods:
```bash
[root@servicenode ~]# kubectl get po
NAME                                             READY   STATUS    RESTARTS   AGE
csi-baremetal-controller-5d7bf6766b-6wjck        4/4     Running   0          16m
csi-baremetal-node-controller-5f4cffcb94-fmfx7   1/1     Running   0          16m
csi-baremetal-operator-5d48b768f-vlbbj           1/1     Running   0          53m
```
4. Check events:
```bash
[root@servicenode ~]# kubectl get events --sort-by=.lastTimestamp | grep ServiceAccount | tail -2
13m         Warning   NodeRoleValidationFailed       deployment/csi-baremetal                              ServiceAccount csi-node-sa has insufficient securityContextConstraints, should have privileged
13m         Warning   ExtenderRoleValidationFailed   deployment/csi-baremetal                              ServiceAccount csi-baremetal-extender-sa has insufficient securityContextConstraints, should have privileged
```
5. Create Role and RoleBinding as described in docs, then check pods:
```bash
[root@servicenode ~]# kubectl get po
NAME                                             READY   STATUS    RESTARTS   AGE
csi-baremetal-controller-5d7bf6766b-6wjck        4/4     Running   0          41m
csi-baremetal-node-45lpz                         4/4     Running   0          21m
csi-baremetal-node-controller-5f4cffcb94-fmfx7   1/1     Running   0          41m
csi-baremetal-node-nj7g5                         4/4     Running   0          21m
csi-baremetal-node-q4shc                         4/4     Running   0          21m
csi-baremetal-operator-5d48b768f-vlbbj           1/1     Running   0          77m
csi-baremetal-se-8wgrb                           1/1     Running   0          21m
csi-baremetal-se-dw79t                           1/1     Running   0          21m
csi-baremetal-se-zl9q8                           1/1     Running   0          21m
```
6. Try to deploy statefulset in default namespace:
```bash
[root@servicenode ~]# kubectl apply -f statefulset.yaml  -n default
service/nginx created
statefulset.apps/web created
```
7. Check deployed pods:
```bash
[root@servicenode ~]# kubectl get po  -n default
NAME    READY   STATUS    RESTARTS   AGE
web-0   1/1     Running   0          4m53s
web-1   1/1     Running   0          3m27s
web-2   1/1     Running   0          119s
```
8. Check volumes:
```bash
[root@servicenode ~]# kubectl get po  -n default
NAME    READY   STATUS    RESTARTS   AGE
web-0   1/1     Running   0          4m53s
web-1   1/1     Running   0          3m27s
web-2   1/1     Running   0          119s
[root@servicenode ~]# kubectl get volumes -n default
NAME                                       SIZE        STORAGE CLASS   HEALTH   CSI_STATUS   LOCATION                               NODE
pvc-0d2a618f-01c1-4cf7-b627-fb02311025c4   104857600   HDDLVG          GOOD     PUBLISHED    d371d89a-c9a5-400b-81fc-b16d7d926f7b   be094edd-76fc-43a5-a698-5d582b4cd0e2
pvc-6b5fbb95-ef83-4890-a77c-6a46ee0755aa   104857600   HDDLVG          GOOD     PUBLISHED    16d428a6-9461-4e57-9051-ff772405c845   f6df20ce-b51d-4859-aea3-bae8601cac94
pvc-9dd1acbd-c565-4d6c-bb57-b10285416617   104857600   HDDLVG          GOOD     PUBLISHED    9b2dad75-fa4c-4270-a3e5-7459bc508f6d   ae4bde04-5f9d-4620-8a1f-40a1fd5db22e
```
